entry_types/scrolled/package/spec/frontend/useEventContextData-spec.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import {useEventContextData} from 'frontend/useEventContextData';
+
+import 'support/fakeBrowserFeatures';
+import {renderHookInEntry} from 'support';
+
+describe('useEventContextData', () => {
+
+  it('returns an object with current section and chapter index', () => {
+
+    const {result} = renderHookInEntry(() => useEventContextData());
+    
+    expect(result.current).toMatchObject({
+      configuration: {
+        title: undefined
```

Comment:
Consider setting up the spec in a way that tests a more realistic case. `renderHookInEntry` allows specifying e. g. a wrapper, which you can maybe use to provide realistic data.

Doing this will be more eloquent in a way of describing a situation in which this hook produces useful data

Reply:
I have tried encapsulating it but it is giving me error. There are some changes in renderHookInEntry like it is not taking wrapper as argument.
Even if I change it to take wrapper as argument it is giving me error.

```const {result} = renderHookInEntry((props) => useEventContextData(), {
      seed: {
        sections: [
          {
            id: 1,
            permaId: 101,
            chapterId: 1,
            position: 1,
            configuration: {
              transition: 'scroll'
            }
          },
        ],
        chapters: [
          {
            id: 1,
            permaId: 10,
            position: 1,
            configuration: {
              title: 'Chapter 1',
              summary: 'An introductory chapter'
            }
          },
        ]
      },
      wrapper: () => ( <EventContextDataProvider sectionIndex={1}/> )
    });```


do you have any idea how can I encapsulate it properly?

Reply:
Indeed, `renderHookInEntry` doesn't accept `wrapper` yet. I think your approach of adding it is the way to go.

We have an example of something similar in scrollPositionLifecycle [1]. Have you tried using a call to a creator function instead of a function literal as the `wrapper` parameter, as we do it there?

[1] https://github.com/codevise/pageflow/blob/f4f4f9523df802c82990bb2ce96fcee5936462fb/entry_types/scrolled/package/spec/support/scrollPositionLifecycle.js#L38

Reply:
Here in this example it is creating custom provider within Test environment.
If I do that that would mean providing hardcoded values like sectionIndex and title and then again testing those
which would become similar to what it is now.

Reply:
It would be different in a way that is crucial for a good test: It would give an example for usage of the interface that will lead to non-default output, and thus explain as well as test this piece of functionality.

Phrased differently, the difference would be: Currently, the spec tests whether there is a hook by the given name that returns data in a specific format regardless of context. After the change we discuss, the test will include a possible valid context, and will test what the hook does in that context. In terms of test quality, I wouldn't call this 'similar to what it is now'

Reply:
`      wrapper: () => ( <EventContextDataProvider sectionIndex={1}/> )`

it is doing same thing. only difference is that it uses actual Provider. 
The error I am getting is related to entryState. that entryState is not defined

Reply:
I think it is more important to test Provider rather than context

Reply:
> ` wrapper: () => ( <EventContextDataProvider sectionIndex={1}/> )`
> 
> it is doing same thing. only difference is that it uses actual Provider.
> The error I am getting is related to entryState. that entryState is not defined

In the example from scrollPositionLifecycle, we don't pass a function literal that returns a wrapper and still needs to be called. Rather, the difference there is that we call a function named `createScrollPositionProvider`, and that function call returns a component `ScrollPositionProvider`. So maybe we could try something along the lines of

```javascript
wrapper: createEventContextDataProvider({sectionIndex: 1, chapterIndex: 2})
```
and
```javascript
function createEventContextDataProvider({sectionIndex, chapterIndex}) {
  return (
    <EventContextDataProvider sectionIndex={sectionIndex} chapterIndex={chapterIndex}/>
  );
}
```
which would mean we also pass a component as `wrapper` and not a function that returns a component when called

Reply:
I am afraid this is not helpful.
wrapper parameter requires a function (which returns react component).
In the example above that you shared it is assigning react component directly to wrapper.

Wrapping it in the function is syntactic sugar. What we actually want is a function which returns react component.
I have tried it by wrapping it inside a function or without it. In both cases I get entryState undefined error.

I suspect I am getting this error because somehow hook is not getting wrapped inside provided wrapper.
That is why it is giving undefined entryState.

Reply:
do you know if there is any place where I can get reference to the seed data structure?

Reply:
> I am afraid this is not helpful.
> wrapper parameter requires a function (which returns react component).
> In the example above that you shared it is assigning react component directly to wrapper.
> 
> Wrapping it in the function is syntactic sugar. What we actually want is a function which returns react component.
> I have tried it by wrapping it inside a function or without it. In both cases I get entryState undefined error.
> 
> I suspect I am getting this error because somehow hook is not getting wrapped inside provided wrapper.
> That is why it is giving undefined entryState.

It shouldn't make a difference, but it is something I still would've tried. Also I agree that if it doesn't work, it isn't helpful. Probably it doesn't make much sense if I try to provide you input for debugging via GitHub comments

Reply:
I am not looking for input on debugging or programming solutions. I need to get clear understanding like e.g. I don't know what should the seed data or inputs to renderHookInEntry look like? 
I tried finding answer to this question in source code but I could not find it.
If I know how renderHookInEntry can be used and its seed data structure then I will be able to solve this issue easily.
Otherwise I am not sure how much I need to look into source code. I am trying to figure out, I am not even sure if what I am looking for is present in source code or not, do you see my problem?

Reply:
> do you know if there is any place where I can get reference to the seed data structure?

You can find it in the documentation of `normalizeSeed` [1], which we use indirectly in `renderHookInEntry`

[1] https://github.com/codevise/pageflow/blob/master/entry_types/scrolled/package/spec/support/normalizeSeed.js

Reply:
> If I know how renderHookInEntry can be used

The way I'd do this would be to look at existing usages of `renderHookInEntry`

Reply:
So the question to which I am trying to find answer for is:

What parameters I can pass to renderHookInEntry, so that my hook useEventContextData get wrapped inside 
the entry?

As it is mentioned in normalizeSeed.js I have tried passing 
```
{
  seed: {
    entry: {}
    sections: [...],
    chapters: [...]
  }
}
```

but still it is giving me error that entryState is undefined

Reply:
> > If I know how renderHookInEntry can be used
> 
> The way I'd do this would be to look at existing usages of `renderHookInEntry`

That was the first thing I did but as I told you renderHookInEntry does not even take wrapper so this is the first use case

Reply:
If you push a branch with your failing spec to your GitHub account, I could have a look

Reply:
sure, please have a look now

Reply:
> > The way I'd do this would be to look at existing usages of `renderHookInEntry`
> 
> That was the first thing I did but as I told you renderHookInEntry does not even take wrapper so this is the first use case

In this case, you could look for usages of `renderInEntry` that receive a wrapper. `renderInEntry` and `renderHookInEntry` are very similar.

Comparing the documentations of `render` [1] and `renderHook` [2], it seems to me that both use a `wrapper` parameter in the same way

[1] https://testing-library.com/docs/react-testing-library/api#wrapper
[2] https://react-hooks-testing-library.com/reference/api#wrapper

Reply:
> > > The way I'd do this would be to look at existing usages of `renderHookInEntry`
> > 
> > 
> > That was the first thing I did but as I told you renderHookInEntry does not even take wrapper so this is the first use case
> 
> In this case, you could look for usages of `renderInEntry` that receive a wrapper. `renderInEntry` and `renderHookInEntry` are very similar.
> 
> Comparing the documentations of `render` [1] and `renderHook` [2], it seems to me that both use a `wrapper` parameter in the same way
> 
> [1] https://testing-library.com/docs/react-testing-library/api#wrapper
> [2] https://react-hooks-testing-library.com/reference/api#wrapper

yes I did that already and that is how I came up with the idea of using wrapper in renderHookInEntry.

Reply:
As you requested in our discussion just now, we'll leave this PR until Tim returns. In that case, no need for me to look into that failing spec, I think